lldb/source/API/SBThread.cpp

```diff
@@ -362,15 +366,15 @@ SBValue SBThread::GetStopReturnValue() {
 
   ValueObjectSP return_valobj_sp;
   std::unique_lock<std::recursive_mutex> lock;
-  ExecutionContext exe_ctx(m_opaque_sp.get(), lock);
+  Process::StopLocker stop_locker;
+  ExecutionContext exe_ctx(m_opaque_sp.get(), lock, stop_locker);
+  if (!stop_locker.IsLocked())
+    return SBValue();
```

Comment:
Shouldn't this return your "Process is not stopped" error SBValue that you made for SBFrame?